zcash_keys/src/keys.rs

```diff
@@ -71,6 +70,33 @@ pub mod sapling {
     }
 }
 
+/// A hash of a seed used for an HD account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct HDSeedFingerprint([u8; BLAKE2B_LENGTH]);
+
+impl HDSeedFingerprint {
+    /// Generates the fingerprint from a given seed.
+    pub fn from_seed(seed: &SecretVec<u8>) -> Self {
+        const PERSONALIZATION: &[u8] = b"Zcash_HD_Seed_FP";
+        let hash = blake2bParams::new()
+            .personal(PERSONALIZATION)
+            .to_state()
+            .update(seed.expose_secret())
+            .finalize();
+        Self(hash.as_array().to_owned())
+    }
+
+    /// Instantiates the fingerprint from a buffer containing a previously computed fingerprint.
+    pub fn from_bytes(hash: [u8; BLAKE2B_LENGTH]) -> Self {
```

Comment:
```suggestion
    pub fn from_bytes(hash: [u8; 32]) -> Self {
```
(We've also had problems with language servers not detecting that a non-exported constant can be replaced with a direct constant, but in this case it is an error to use the constant at all.)